packages/orchestration/src/examples/auto-stake-it-tap-kit.js

```diff
@@ -0,0 +1,155 @@
+ */
+const prepareMakeStakingTapKit = (zone, { watch }) => {
+  return zone.exoClassKit(
+    'StakingTap',
+    {
+      tap: M.interface('AutoStakeItTap', {
+        receiveUpcall: M.call(M.record()).returns(
+          M.or(VowShape, M.undefined()),
+        ),
+      }),
+      transferWatcher: M.interface('TransferWatcher', {
+        onFulfilled: M.call(M.undefined())
+          .optional(M.bigint())
+          .returns(VowShape),
+      }),
+    },
+    /** @param {StakingTapState} initialState */
+    initialState => {
+      mustMatch(initialState, StakingTapStateShape);
+      return /** @type {StakingTapState} */ (harden(initialState));
+    },
+    {
+      tap: {
+        /**
+         * Transfers from localAccount to stakingAccount, then delegates from
+         * the stakingAccount to `validator` if the expected token (remoteDenom)
+         * is received.
+         *
+         * @param {VTransferIBCEvent} event
+         */
+        receiveUpcall(event) {
+          trace('receiveUpcall', event);
+
+          // ignore packets from unknown channels
+          if (event.packet.source_channel !== this.state.sourceChannel) {
+            return;
+          }
+
+          const tx = /** @type {FungibleTokenPacketData} */ (
+            JSON.parse(atob(event.packet.data))
+          );
+          trace('receiveUpcall packet data', tx);
+
+          const { remoteDenom, localChainAddress } = this.state;
+          // ignore outgoing transfers
+          if (tx.receiver !== localChainAddress.value) {
+            return;
+          }
+          // only interested in transfers of `remoteDenom`
+          if (tx.denom !== remoteDenom) {
+            return;
+          }
+
+          const { localAccount, localDenom, remoteChainAddress } = this.state;
+          return watch(
+            E(localAccount).transfer(
+              {
+                denom: localDenom,
+                value: BigInt(tx.amount),
+              },
+              remoteChainAddress,
+            ),
+            this.facets.transferWatcher,
+            BigInt(tx.amount),
+          );
+        },
+      },
+      transferWatcher: {
+        /**
+         * @param {void} _result
+         * @param {bigint} value the qty of uatom to delegate
+         */
+        onFulfilled(_result, value) {
+          const { stakingAccount, validator, remoteDenom } = this.state;
+          return watch(
+            E(stakingAccount).delegate(validator, {
+              denom: remoteDenom,
+              value,
+            }),
+          );
+        },
+      },
+    },
+  );
+};
+
+/**
+ * Provides a {@link TargetApp} that reacts to an incoming IBC transfer by:
+ *
+ * 1. transferring the funds to the staking account specified at initialization
+ * 2. delegating the funds to the validator specified at initialization
+ *
+ * XXX consider a facet with a method for changing the validator
+ *
+ * XXX consider logic for multiple stakingAccounts + denoms
```

Comment:
just checking that these aren't requirements for MVP (in which case they'd be FIXME)